rest/oidc_api_test.go

```diff
@@ -1,139 +1,816 @@
 package rest
 
 import (
-	"io"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
 	"net/http"
 	"net/http/httptest"
 	"net/url"
+	"regexp"
 	"strings"
 	"testing"
+	"time"
 
 	"github.com/couchbase/sync_gateway/auth"
+	"github.com/couchbase/sync_gateway/base"
+	"github.com/gorilla/mux"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"gopkg.in/square/go-jose.v2/jwt"
 )
 
+// A wantErrKind represents the specific error expected.
+type wantError struct {
+
+	// kind represents the specific kind of wantError expected.
+	kind wantErrKind
+
+	// code represents the expected status code from HTTP response.
+	code int
```

Comment:
In general it's preferable to put single-line comments on the same line as the property, and to avoid redundant language in the comment
```suggestion
	expectedErrorCode int  // Expected HTTP response code
```

Reply:
Added single line comments against these fields.